docs/interactions/Message_Components.md

```diff
@@ -418,3 +423,50 @@ Select menus support single-select and multi-select behavior, meaning you can pr
     "version": 1
 }
 ```
+
+## Text Inputs
+
+Text inputs are an interactive component that render on modals. They can be used to collect short-form or long-form text.
+
+###### Text Input Example
+
+```json
+// this is a modal
+{
+  "title": "My Cool Modal",
+  "custom_id": "cool_modal",
+  "components": [{
+    "type": 1,
+    "components": [{
+      "type": 4,
+      "custom_id": "name",
+      "label": "Name",
+      "style": 1,
+      "min_length": 1,
+      "max_length": 4000,
+      "placeholder": "John",
+      "required": true
+    }]
+  }]
+}
+```
+
+###### Text Input Structure
+
+| Field        | Type    | Description                                                                                 |
+| ------------ | ------- | ------------------------------------------------------------------------------------------- |
+| type         | integer | `4` for a text input                                                                        |
+| custom_id    | string  | a developer-defined identifier for the input, max 100 characters                            |
+| style        | integer | the [Text Input Style](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/text-inputs-text-input-styles) |
+| label?       | string  | the label for this component                                                                |
```

Comment:
Why is this optional? The notion site doesn't state it is
![image](https://user-images.githubusercontent.com/67214928/153057621-c726b7af-591c-46f8-b71b-c1eaf90cba67.png)